examples/own_dataset/train.py

```diff
@@ -139,48 +166,43 @@ def postprocess_label(label_list):
     else:
         raise ValueError('[ERROR] Invalid method {}'.format(method))
 
-    train_iter = I.SerialIterator(train, args.batchsize)
-    val_iter = I.SerialIterator(val, args.batchsize,
+    train_iter = iterators.SerialIterator(train, args.batchsize)
+    val_iter = iterators.SerialIterator(val, args.batchsize,
                                 repeat=False, shuffle=False)
 
-    def scaled_abs_error(x0, x1):
-        if isinstance(x0, Variable):
-            x0 = cuda.to_cpu(x0.data)
-        if isinstance(x1, Variable):
-            x1 = cuda.to_cpu(x1.data)
-        if args.scale == 'standardize':
-            scaled_x0 = ss.inverse_transform(cuda.to_cpu(x0))
-            scaled_x1 = ss.inverse_transform(cuda.to_cpu(x1))
-            diff = scaled_x0 - scaled_x1
-        elif args.scale == 'none':
-            diff = cuda.to_cpu(x0) - cuda.to_cpu(x1)
-        return numpy.mean(numpy.absolute(diff), axis=0)[0]
-
-    classifier = L.Classifier(model, lossfun=F.mean_squared_error,
-                              accfun=scaled_abs_error)
+    regressor = Regressor(
+        model, lossfun=F.mean_squared_error,
+        metrics_fun={'abs_error': ScaledAbsError(scale=args.scale, ss=ss)},
+        device=args.gpu)
 
-    if args.gpu >= 0:
-        chainer.cuda.get_device_from_id(args.gpu).use()
-        classifier.to_gpu()
-
-    optimizer = O.Adam()
-    optimizer.setup(classifier)
+    optimizer = optimizers.Adam()
+    optimizer.setup(regressor)
 
     updater = training.StandardUpdater(train_iter, optimizer, device=args.gpu,
                                        converter=concat_mols)
     trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
-    trainer.extend(E.Evaluator(val_iter, classifier, device=args.gpu,
+    trainer.extend(E.Evaluator(val_iter, regressor, device=args.gpu,
                                converter=concat_mols))
     trainer.extend(E.snapshot(), trigger=(args.epoch, 'epoch'))
     trainer.extend(E.LogReport())
-    # Note that scaled errors are reported as (validation/)main/accuracy
-    trainer.extend(E.PrintReport(['epoch', 'main/loss', 'main/accuracy',
+    # Note that original scale absolute errors are reported in
+    # (validation/)main/abs_error
+    trainer.extend(E.PrintReport(['epoch', 'main/loss', 'main/abs_error',
                                   'validation/main/loss',
-                                  'validation/main/accuracy',
+                                  'validation/main/abs_error',
                                   'elapsed_time']))
     trainer.extend(E.ProgressBar())
     trainer.run()
 
+    # --- save regressor's parameters ---
+    protocol = args.protocol
+    save_model_path = os.path.join(args.out, 'model.npz')
```

Comment:
It would be my taste but `save_model_path` looks to me that it indicates an act of saving a path to a model. How about `model_path`?

Reply:
done